vectara_agentic/tools.py

```diff
@@ -251,12 +251,16 @@ def _build_filter_string(kwargs: Dict[str, Any], tool_args_type: Dict[str, str])
                         # = and != operators can be numeric or string
                         if rhs.isdigit() or is_float(rhs):
                             filter_parts.append(f"{prefix}.{key}{matched_operator}{rhs}")
+                        elif rhs.lower() in ["true", "false"]:
```

Comment:
It is logical to assume that a value of "true" or "false" implies that the filter attribute is a Boolean. So I think that this logic is fine. But I can imagine an edge case where there is a Text typed filter attribute that accepts strings for "true" or "false". 

So short of actually loading the corpus schema to definitively determine the type of the attribute, I think that this approach is fine. But I would add documentation somewhere to explicitly call out the assumption that having a value of "true" or "false" means that we treat the filter attribute as a Boolean.

Reply:
Yes, we can do that after migration to APIv2